exporter/sapmexporter/exporter.go

```diff
@@ -126,17 +124,26 @@ func (se *sapmExporter) pushTraceData(ctx context.Context, td ptrace.Traces) err
 	return nil
 }
 
-func (se *sapmExporter) retrieveAccessToken(md ptrace.ResourceSpans) string {
+func (se *sapmExporter) retrieveAccessToken(ctx context.Context, md ptrace.ResourceSpans) string {
 	if !se.config.AccessTokenPassthrough {
 		// Nothing to do if token is pass through not configured or resource is nil.
 		return ""
 	}
 
-	attrs := md.Resource().Attributes()
-	if accessToken, ok := attrs.Get(splunk.SFxAccessTokenLabel); ok {
-		return accessToken.Str()
+	var token string
+	switch {
+	case ctx.Value("X-SF-TOKEN") != nil:
```

Comment:
Please use a `SFxAccessTokenHeader` constant defined in internal/splunk instead of hardcoded value